BlueprintUI/Sources/Layout/SingleChildLayout.swift

```diff
@@ -21,3 +29,164 @@ public protocol SingleChildLayout {
+
+    /// Returns the size of the element, given a proposed size constraint and the container's
+    /// subelement.
+    ///
+    /// Implement this method to tell your custom layout container’s parent how much space the
+    /// container needs for a subelement, given a size constraint. The parent might call this method
+    /// more than once during a layout pass with different proposed sizes to test the flexibility of
+    /// the container.
+    ///
+    /// In Blueprint, parents ultimately choose the size of their children, so the actual size that
+    /// this container is laid out in may not be a size that was returned from this method.
+    ///
+    /// ## Sizing Rules
+    ///
+    /// For performance reasons, the layout engine may deduce the measurement of your container for
+    /// some constraint values without explicitly calling
+    /// ``sizeThatFits(proposal:subelement:cache:)``. To ensure that the deduced value is correct,
+    /// your layout must follow some ground rules:
+    ///
+    /// 1. **Given one fixed constraint axis, the element's growth along the other axis should be
+    ///    _monotonic_.** That is, an element can grow when given a larger constraint, or shrink
+    ///    when given a smaller constraint, but it should never shrink when given a larger
+    ///    constraint. When growing on one axis, it is OK to shrink along the other axis, such as a
+    ///    block of text that re-flows as the width changes.
+    /// 2. If your element has no intrinsic size along an axis, you can represent that in a couple
+    ///     ways:
+    ///     - You can return a fixed value representing the minimum size for your element. In this
+    ///       approach, a containing element is usually responsible for stretching your element to
+    ///       fill desired space.
+    ///     - You can return a size that entirely fills the constraint proposed. In this approach,
+    ///       you **must** return `.infinity` when the constraint is
+    ///       ``SizeConstraint/Axis/unconstrained``. Otherwise, your behavior would be in violation
+    ///       of rule #1.
+    ///
+    /// - Parameters:
+    ///   - proposal: A size constraint for the container. The container's parent element that calls
+    ///     this method might call the method more than once with different constraints to learn
+    ///     more about the container’s flexibility before choosing a size for placement.
+    ///   - subelement: A proxy that represents the element that the container arranges. You can use
+    ///     the proxy to get information about the subelement as you determine how much space the
+    ///     container needs to display it.
+    ///   - cache: Optional storage for calculated data that you can share among the methods of your
+    ///     custom layout container. See ``makeCache(subelement:)-33y4l`` for details.
+    /// - Returns: A size that indicates how much space the container needs to arrange its
+    ///   subelement.
```

Comment:
Do we need to copy this again? It might be best to reference the previous block.

Reply:
> Do we need to copy this again? It might be best to reference the previous block.

Do you mean replacing these docs with a link to the analogous calls in `Layout`? I just didn't want to make folks jump through a link to read docs. I don't feel strongly about it though.

Reply:
I think it may be worth asking them to jump through rather than try to keep this large body of text up to date in multiple locations. That said, I don't feel strongly about it, nor expect it to change much.

Reply:
OK, I see what you mean. Kept the summary and removed everything "below the fold" of Xcode's quick help [here](https://github.com/square/Blueprint/pull/446/commits/1d3548120da3fd07b63d39871c136244a88c757c#diff-4bc18050c5802990f199763152c93e481d5eafdc7911ace7a6719cfb8bad11e4).